examples/libmctp.rs

```diff
@@ -0,0 +1,111 @@
+#![no_std]
+/// This is a example of a MCTP master device
+use libmctp::smbus::{MCTPSMBusContext, VendorIDFormat};
+use libtock::i2c_master::I2cBuffer;
+use libtock::result::TockResult;
+use libtock::syscalls;
+use libtock::println;
+
+libtock_core::stack_size! {0x800}
+
+// The address of this device
+const MY_ID: u8 = 0x23;
+const DEST_ID: u8 = 0x10;
+// Support vendor defined protocol 0x7E
+const MSG_TYPES: [u8; 1] = [0x7E];
+// Specify a PCI vendor ID that we support
+const VENDOR_IDS: [libmctp::smbus::VendorIDFormat; 1] = [VendorIDFormat {
+    // PCI Vendor ID
+    format: 0x00,
+    // PCI VID
+    data: 0x1414,
+    // Extra data
+    numeric_value: 4,
+}];
+
+#[libtock::main]
+async fn main() -> TockResult<()> {
+    let mut drivers = libtock::retrieve_drivers()?;
+    drivers.console.create_console();
+    println!("Starting libmctp example");
+    let i2c_driver = drivers.i2c.init_driver()?;
+
+    // Check that I2C exists
+    let present = i2c_driver.check_present();
+
+    match present {
+        Ok(_) => {
+            println!("Found the I2C device");
+        }
+        Err(_) => {
+            println!("No I2C device, yielding");
+            loop {
+                unsafe { syscalls::raw::yieldk() };
+            }
+        }
+    }
```

Comment:
The `match` on lines 36 - 46 may be better expressed as:

```rust
if present.is_err() {
    println!("No I2C device, yielding");
    loop {
        unsafe { syscalls::raw::yieldk() };
    }
}
println!("Found the I2C device");
```

I'd suggest inlining `i2c_driver.check_present()` directly into the if statement as well (i.e. remove the `present` variable, as it is only used once).